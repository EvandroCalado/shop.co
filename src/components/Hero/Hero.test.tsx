import { render, screen } from '@testing-library/react';
import { Hero } from '.';

describe('<Hero />', () => {
  it('Shroud render with default value', () => {
    render(<Hero />);

    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument();
    expect(screen.getByText(/Browse through/i)).toBeInTheDocument();
    expect(
      screen.getByRole('button', { name: 'shop now' }),
    ).toBeInTheDocument();
    expect(screen.getByRole('heading', { name: '200+' })).toBeInTheDocument();
    expect(screen.getByRole('heading', { name: '2,000+' })).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { name: '30,000+' }),
    ).toBeInTheDocument();
    expect(screen.getByRole('img')).toBeInTheDocument();
  });

  it('Shroud render a snapshot', () => {
    const { container } = render(<Hero />);

    expect(container).toMatchSnapshot();
  });
});
