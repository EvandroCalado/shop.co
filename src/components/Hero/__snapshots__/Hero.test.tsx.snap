// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Hero /> > Shroud render a snapshot 1`] = `
<div>
  <section
    class="flex flex-col items-center justify-between bg-[#f0f0f0] px-6 md:px-16 lg:flex-row lg:px-24"
  >
    <div
      class="space-y-10 py-6 lg:flex-1"
    >
      <h1
        class="text-4xl font-black uppercase text-black/95 md:text-6xl"
      >
        find clothes
        <br />
         that matches
        <br />
         your style
      </h1>
      <p
        class="max-w-xl text-sm text-zinc-500"
      >
        Browse through our diverse range of meticulously crafted garments, designed to bring out your individuality and cater to your sense of style.
      </p>
      <button
        class="bg-black/95 text-white hover:bg-black/85 rounded-full px-14 py-3 text-sm capitalize transition-all duration-150 w-full md:w-auto"
      >
        shop now
      </button>
      <div
        class="flex flex-wrap items-center gap-4 border-black md:gap-8 [&>div:not(:last-child)]:border-r"
      >
        <div
          class="pr-4 md:pr-8"
        >
          <h4
            class="text-2xl font-bold md:text-4xl"
          >
            200+
          </h4>
          <span
            class="text-[12px] text-zinc-500"
          >
            international brands
          </span>
        </div>
        <div
          class="pr-4 md:pr-8"
        >
          <h4
            class="text-2xl font-bold md:text-4xl"
          >
            2,000+
          </h4>
          <span
            class="text-[12px] text-zinc-500"
          >
            high-quality products
          </span>
        </div>
        <div
          class="pr-4 md:pr-8"
        >
          <h4
            class="text-2xl font-bold md:text-4xl"
          >
            30,000+
          </h4>
          <span
            class="text-[12px] text-zinc-500"
          >
            happy customers
          </span>
        </div>
      </div>
    </div>
    <div
      class="relative h-[380px] w-full sm:h-[480px] md:h-[750px] lg:flex-1"
    >
      <img
        alt="logo"
        class="object-cover"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=%2Fhero.png&w=3840&q=75"
        srcset="/_next/image?url=%2Fhero.png&w=640&q=75 640w, /_next/image?url=%2Fhero.png&w=750&q=75 750w, /_next/image?url=%2Fhero.png&w=828&q=75 828w, /_next/image?url=%2Fhero.png&w=1080&q=75 1080w, /_next/image?url=%2Fhero.png&w=1200&q=75 1200w, /_next/image?url=%2Fhero.png&w=1920&q=75 1920w, /_next/image?url=%2Fhero.png&w=2048&q=75 2048w, /_next/image?url=%2Fhero.png&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
  </section>
</div>
`;
